uni-stark/src/prover.rs

```diff
@@ -63,49 +204,117 @@ where
+    let qc_domains = quotient_domain.split_domains(num_quotient_chuncks);
+
+    // TODO: This treats the all `q_ij` as if they are evaluations over the domain `H`.
+    // This doesn't matter for low degree-ness but we need to be careful when checking
+    // equalities.
+
+    // When computing quotient_data, we take `q_ij` (defined on `gj H`), re-interpret it to
+    // be defined on `H`, e.g. replacing it with `q_ij'(x) = q_ij(gj x)`. Then we do a coset LDE to
+    // get it's evaluations on `(g/gj) H'` which we commit to. Then when computing opening values
+    // we again re-interpret the evaluation vector to be defined on `g H` e.g. replacing our
+    // polynomial with `q_ij''(x) = q_ij'(gj^{-1} x) = q_ij(gj gj^{-1} x) = x`.
+    // In other words our commitment does actually compute the evaluations of `q_ij(x)` over `gH`.
+    // Despite seemingly doing something else...
+
+    // For each polynomial `q_ij`, compute the evaluation vector of `q_ij(x)` over `gH'`. This
+    // is then hashed into a Merkle tree with it's rows bit-reversed.
+    //      quotient_commit contains the root of the tree
+    //      quotient_data contains the entire tree.
+    //          - quotient_data.leaves is a pair of matrices containing the `q_i0(x)` and `q_i1(x)`.
     let (quotient_commit, quotient_data) = info_span!("commit to quotient poly chunks")
         .in_scope(|| pcs.commit(izip!(qc_domains, quotient_chunks).collect_vec()));
     challenger.observe(quotient_commit.clone());
 
+    // Combine our commitments to the trace and quotient polynomials into a single object which
+    // will be passed to the verifier.
     let commitments = Commitments {
         trace: trace_commit,
         quotient_chunks: quotient_commit,
     };
 
+    // Get the second Fiat-Shamir challenge, `zeta`, an opening point.
+    // Along with `zeta_next = next(zeta)` where next is the unique successor linear function
+    // on the initial domain. In the usual STARK case, this is `h * zeta`.
+    //
+    // TODO: What is this in the Circle STARK case?
     let zeta: SC::Challenge = challenger.sample();
-    let zeta_next = trace_domain.next_point(zeta).unwrap();
+    let zeta_next = initial_trace_domain.next_point(zeta).unwrap();
 
+    // The prover opens the trace polynomials at `zeta` and `zeta_next` and the
+    // and quotient polynomials at `zeta`.
+    //
+    // TODO: What are opened_values, opening_proof??
+    // This also produces a FRI proof??
+    // Why is zeta the right point to evaluate quotient_data at?
     let (opened_values, opening_proof) = info_span!("open").in_scope(|| {
         pcs.open(
             vec![
                 (&trace_data, vec![vec![zeta, zeta_next]]),
                 (
                     &quotient_data,
                     // open every chunk at zeta
-                    (0..quotient_degree).map(|_| vec![zeta]).collect_vec(),
+                    iter::repeat_n(vec![zeta], num_quotient_chuncks).collect_vec(),
```

Comment:
Simpler still:
```suggestion
                    vec![vec![zeta]; num_quotient_chuncks],
```
(And you can remove `use core::iter` above.)